src/main/java/io/reactivex/Notification.java

```diff
@@ -16,52 +16,226 @@
 import io.reactivex.internal.functions.Objects;
 
 /**
- * Utility class to help construct notification objects.
+ * Represents the reactive signal types: onNext, onError and onComplete and
+ * holds their parameter values (a value, a Throwable, nothing).
+ * @param <T> the value type
  */
-public final class Notification {
-    static final Try<Optional<Object>> COMPLETE = Try.ofValue(Optional.<Object>empty());
-    
-    private Notification() {
-        throw new IllegalStateException();
+public abstract class Notification<T> {
+    /** Not meant to be implemented externally. */
+    Notification() {
+        
     }
     
-    @SuppressWarnings({ "rawtypes", "unchecked" })
-    public static <T> Try<Optional<T>> complete() {
-        return (Try)COMPLETE;
-    }
+    /**
+     * Returns true if this notification is an onComplete signal.
+     * @return true if this notification is an onComplete signal
+     */
+    public abstract boolean isOnComplete();
+    
+    /**
+     * Returns true if this notification is an onError signal and
+     * {@link #getError()} returns the contained Throwable.
+     * @return true if this notification is an onError signal
+     * @see #getError()
+     */
+    public abstract boolean isOnError();
+    
+    /**
+     * Returns true if this notification is an onNext signal and
+     * {@link #getValue()} returns the contained value.
+     * @return true if this notification is an onNext signal
+     * @see #getValue()
+     */
+    public abstract boolean isOnNext();
+    
+    /**
+     * Returns the contained value if this notification is an onNext
+     * signal, null otherwise.
+     * @return the value contained or null
+     * @see #isOnNext()
+     */
+    public abstract T getValue();
     
-    @SuppressWarnings({ "rawtypes", "unchecked" })
-    public static <T> Try<Optional<T>> error(Throwable e) {
-        return (Try)Try.ofError(e);
+    /**
+     * Returns the container Throwable error if this notification is an onError
+     * signal, null otherwise.
+     * @return the Throwable error contained or null
+     * @see #isOnError()
+     */
+    public abstract Throwable getError();
+    
+    /**
+     * Constructs an onNext notification containing the given value.
+     * @param <T> the value type
+     * @param value the value to carry around in the notification, not null
+     * @return the new Notification instance
+     * @throws NullPointerException if value is null
+     */
+    public static <T> Notification<T> createOnNext(T value) {
+        Objects.requireNonNull(value, "value is null");
```

Comment:
Yes, rule §2.13 forbids the use of null values, so do we on all possible entry points.